system/extensions/esp32/firmware.toit

```diff
@@ -97,6 +126,15 @@ class FirmwareWriter extends ServiceResource:
         written_ = ota_write_ buffer_
         fullness_ = 0
 
+  pad size/int value/int -> int:
+    return List.chunk_up 0 size (buffer_.size - fullness_) buffer_.size: | from to chunk |
+      end := fullness_ + chunk
+      buffer_.fill --from=fullness_ --to=end value
+      fullness_ = end
+      if fullness_ == buffer_.size:
```

Comment:
I don't understand the test:
The chunk-up method should make sure that the buffer is filled. The only case where this might not be true, is for the last chunk. Does that mean that we don't want to write the last chunk if it doesn't fully fill the buffer? if yes, add a comment.

Reply:
Will add comment in follow up PR.